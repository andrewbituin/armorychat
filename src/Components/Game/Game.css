.game-title {
  font-size: 20px;
  margin: 0;
  display: inline-block;
  margin: 0 auto;
  border: 2px solid transparent;
}
  
.game-item-container{
  display: inline-block;
  margin: 5px 20px;
}

.game-item-container a {
  display: inline-flex;
  justify-content: center;
  flex-direction: column;
  text-decoration: none;
  color: white;
}

.game-item-container a:hover .game-title {
  border-bottom: 2px solid #fff;
}

.game-item-container img {
  width: 200px;
  box-shadow: 0 5px 13px 0 #00000080;
  border-radius: 2px;
  transition: 0.15s ease;
}

.game-item-container a:hover img {
  width: 206px;
  transition: 0.15s ease;
  box-shadow: 0 5px 26px 0 #000000B3;
}

.game-img{
  display: flex;
  align-items: center;
  justify-content: center;
  width: 220px;
  height: 220px;
}

.game-item-container span, .game-item-container p {
  font-size: 14px;
  font-weight: normal;
}

.game-item-container p {
  margin-top: 3px;
  margin-bottom: 8px;
  text-align: center;
}

.game-tags-container {
  margin: 0 auto;
  display: flex;
  justify-content: center;
  align-content: space-around;
  flex-wrap: wrap;
}

.game-tags-container span {
  margin: 0 5px 5px !important;
}
  
@media screen and (max-width: 500px){
  .game-item-container {
    margin: 0 auto;
  }
}