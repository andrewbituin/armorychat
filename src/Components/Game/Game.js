import React from "react";
import { Link } from "react-router-dom";
import './Game.css';
import config from '../../config';

export default function Game(props) {
  let data = props.props;
  return (
    <li className='game-item-container'>
      <Link to={`/games/${data.id}`}>
      <div className='game-img'>
        <img src={`${config.IMAGES_ENDPOINT}/${data.id}/Main-Image.png`} alt="Game Cover" />
      </div>
        <h2 className="game-title">{data.title}</h2>
      </Link>
      <p>{data.party_count} {data.party_count !== 1 ? 'Squads' : 'Squad'}</p>
      <div className="game-tags-container">
        {data.tags.map((tag, i) => {
          return <span className='small-detail' key={i}>{tag}</span>;
        })}
      </div>
    </li>
  )
}

