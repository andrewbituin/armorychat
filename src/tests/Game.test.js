import React from "react";
import Game from "../Components/Game/Game";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

const data = {
  id: "aa0e8ce9-1a71-42e7-804d-6838556fa6ed",
  title: "Test Game",
  image_url:
    "https://res.cloudinary.com/squadarmory/image/upload/v1557172650/overwatch_epjgky.png",
  tags: ["Test tag1", "test tag2"]
};

describe("Game component", () => {
  it("renders without crashing", () => {
    shallow(<Game props={data} />);
  });

  it("renders the UI as expected", () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Game props={data}/>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
