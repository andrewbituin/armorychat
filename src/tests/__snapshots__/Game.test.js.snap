// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game component renders the UI as expected 1`] = `
<li
  className="game-item-container"
>
  <a
    href="/games/aa0e8ce9-1a71-42e7-804d-6838556fa6ed"
    onClick={[Function]}
  >
    <div
      className="game-img"
    >
      <img
        alt="Game Cover"
        src="https://limitless-brushlands-45977.herokuapp.com/images/aa0e8ce9-1a71-42e7-804d-6838556fa6ed/Main-Image.png"
      />
    </div>
    <h2
      className="game-title"
    >
      Test Game
    </h2>
  </a>
  <p>
     
    Squads
  </p>
  <div
    className="game-tags-container"
  >
    <span
      className="small-detail"
    >
      Test tag1
    </span>
    <span
      className="small-detail"
    >
      test tag2
    </span>
  </div>
</li>
`;
